### PR TITLE
lookup templated value of a var

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

title says all

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

lookup/vars
##### ANSIBLE VERSION

```
2.5
```

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure in multiple integration tests due to `The conditional check 'result.failed == false' failed. The error was: error while evaluating conditional (result.failed == false): 'dict object' has no attribute 'failed'`.

Unit tests also failed due to:

```
>       self.assertIsInstance(res[0], Block)
E       AssertionError: TASK: include is not an instance of <class 'ansible.playbook.block.Block'>

test/units/playbook/test_helpers.py:254: AssertionError
```

```
>                               variable_manager=self.mock_variable_manager, loader=self.fake_include_loader)
E       AssertionError: AnsibleParserError not raised by load_list_of_tasks

test/units/playbook/test_helpers.py:245: AssertionError
```

**Comment by mattclay** [comment]
CI failure in integration tests. Here's one of the errors: `The task includes an option with an undefined variable. The error was: 'item' is undefined\n\nThe error appears to have been in '/root/ansible/test/integration/targets/docker_secret/tasks/Fedora.yml': line 1, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Install Docker pre-reqs\n  ^ here\n"`

**Comment by mattclay** [comment]
CI failure same as before.

**Comment by bcoca** [comment]
nope, diff file

File: lib/ansible/playbook/task.py [hunk 1]
```diff
@@ -215,7 +215,7 @@ def preprocess_data(self, ds):
             if k in ('action', 'local_action', 'args', 'delegate_to') or k == action or k == 'shell':
                 # we don't want to re-assign these values, which were determined by the ModuleArgsParser() above
                 continue
-            elif k.replace("with_", "") in lookup_loader:
+            elif k.startswith('with_') and k.replace("with_", "") in lookup_loader:
```
Comment (maxamillion) [inline lib/ansible/playbook/task.py:218]: Why is this change needed for the addition of the var lookup? I don't understand the context, any chance we could add it to the comment?

File: test/integration/targets/docker_secret/tasks/Fedora.yml [hunk 2]
```diff
@@ -2,7 +2,7 @@
   dnf:
     name: "{{ item }}"
     state: present
-  items:
+  loop:
```
Comment (maxamillion) [inline test/integration/targets/docker_secret/tasks/Fedora.yml:5]: Why change this? Why change this? Seems out of scope.

File: test/integration/targets/docker_secret/tasks/RedHat.yml [hunk 3]
```diff
@@ -5,7 +5,7 @@
   yum:
     name: "{{ item }}"
     state: present
-  items:
+  loop:
```
Comment (maxamillion) [inline test/integration/targets/docker_secret/tasks/RedHat.yml:8]: Why change this? Seems out of scope.

**Review by maxamillion — COMMENTED** [review]
Code looks good for the most part, but was curious about the motivation behind a couple changes.

File: lib/ansible/playbook/task.py (hunk 1 above)
Comment (bcoca) [reply lib/ansible/playbook/task.py:218]: vars let me discover we were doing something improper and allowing direct use of the lookup as a keyword, since 'vars' already exists ... it was getting overloaded, this fixes that bug.

File: test/integration/targets/docker_secret/tasks/Fedora.yml (hunk 2 above)
Comment (bcoca) [reply test/integration/targets/docker_secret/tasks/Fedora.yml:5]: these are examples of tests relying on the bug described above

Comment (maxamillion) [reply test/integration/targets/docker_secret/tasks/Fedora.yml:5]: :+1:

File: lib/ansible/playbook/task.py (hunk 1 above)
Comment (maxamillion) [reply lib/ansible/playbook/task.py:218]: :+1:
